homeassistant/components/epion/sensor.py

```diff
@@ -0,0 +1,137 @@
+"""Support for Epion API."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    PERCENTAGE,
+    UnitOfPressure,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import EpionCoordinator
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        key="co2",
+        name="CO2",
+        suggested_display_precision=0,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        key="temperature",
+        name="Temperature",
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        key="humidity",
+        name="Humidity",
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.ATMOSPHERIC_PRESSURE,
+        native_unit_of_measurement=UnitOfPressure.HPA,
+        key="pressure",
+        name="Pressure",
+        suggested_display_precision=0,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add an Epion entry."""
+    coordinator: EpionCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = [
+        EpionSensor(coordinator, epion_device, description)
+        for epion_device in coordinator.data.values()
+        for description in SENSOR_TYPES
+    ]
+
+    async_add_entities(entities)
+
+
+class EpionSensor(CoordinatorEntity, SensorEntity):
+    """Representation of an Epion Air sensor."""
+
+    def __init__(
+        self,
+        coordinator: EpionCoordinator,
+        epion_device: dict,
```

Comment:
Can we instead pass in the device id? We can then create a property in here like

```py
@property
def device() -> dict[..]:
    return self.coordinator.data[self.device_id]
```

To have easy access to the device